x/ccv/provider/keeper/staking_keeper_interface.go

```diff
@@ -0,0 +1,81 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Iterates over the consensus-active validators by power.
+// The same as IterateBondedValidatorsByPower in the StakingKeeper,
+// but only returns the first MaxProviderConsensusValidators validators.
+// This is used to implement the interface of the staking keeper to interface with
+// modules that need to reference the consensus-active validators.
+func (k Keeper) IterateBondedValidatorsByPower(ctx context.Context, fn func(index int64, validator stakingtypes.ValidatorI) (stop bool)) error {
+	maxProviderConsensusVals := k.GetMaxProviderConsensusValidators(sdk.UnwrapSDKContext(ctx))
+	counter := int64(0)
+	return k.stakingKeeper.IterateBondedValidatorsByPower(ctx, func(index int64, validator stakingtypes.ValidatorI) (stop bool) {
+		if counter >= maxProviderConsensusVals {
+			return true
+		}
+		counter++
+		return fn(index, validator)
+	})
+}
+
+// Gets the amount of bonded tokens, which is equal
+// to the amount of tokens of the consensus-active validators.
+// The same as TotalBondedTokens, but only counts
+// tokens of the first MaxProviderConsensusValidators validators.
+// This is used to implement the interface of the staking keeper to interface with
+// modules that need to reference the consensus-active validators.
+func (k Keeper) TotalBondedTokens(ctx context.Context) (math.Int, error) {
```

Comment:
Looking at [`TotalBondedTokens`](https://github.com/cosmos/cosmos-sdk/blob/af357c8d301e0093bdb7a5bc3b8706cf9381b9fa/x/staking/keeper/pool.go#L80) it's not fully clear to me that bonded tokens does not include the tokens of non-consensus-active validators.

Reply:
I changed the docstring, does that help?
Or are you talking about the code?

Reply:
Here are some references for how the original TotalBondedTokens function behaves:

It just gets the tokens in the bonded pool here https://github.com/cosmos/cosmos-sdk/blob/a32186608aab0bd436049377ddb34f90006fcbf7/x/staking/keeper/pool.go#L80

the bonded pool contains the tokens from bonded validators. e.g. see here that tokens are sent to the bonded pool when someone delegates to an active validator https://github.com/cosmos/cosmos-sdk/blob/a32186608aab0bd436049377ddb34f90006fcbf7/x/staking/keeper/delegation.go#L913,
and how tokens are transferred between bonded/not-bonded pool on valstatechanges https://github.com/cosmos/cosmos-sdk/blob/a32186608aab0bd436049377ddb34f90006fcbf7/x/staking/keeper/val_state_change.go#L251